File: BBGestureBack_Full_oc/Classes/Main/BBUserGuider/BBUserGuiderController.m
//  希望您的举手之劳，能为我点颗赞，谢谢~
//  代码地址: https://github.com/Bonway/BBGestureBack
//  BBGestureBack
//  Created by Bonway on 2016/3/17.
//  Copyright © 2016年 Bonway. All rights reserved.
//

#import "BBUserGuiderController.h"
#import "AppDelegate.h"
#import "BBGestureBack.h"
@interface BBUserGuiderController ()<UIScrollViewDelegate>

@property (nonatomic,strong)UIScrollView  *userGuiderScrollView;
@property (nonatomic,strong)UIButton  *startBtn;
@property (nonatomic,weak)UIImageView   *userGuiderImgView;

@end

@implementation BBUserGuiderController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self.view addSubview:self.userGuiderScrollView];
}

- (UIScrollView *)userGuiderScrollView{
    if (!_userGuiderScrollView) {
        _userGuiderScrollView=[[UIScrollView  alloc] initWithFrame:self.view.bounds];
        
        NSString *fileStr = [[NSBundle mainBundle] pathForResource:@"BBUserGuider" ofType:@"bundle"];
        NSBundle *imageBundle=[NSBundle bundleWithPath:fileStr];
        
        for (int i=0; i<kBBUserGuiderImgCount; i++) {
            UIImageView  *imageView=[[UIImageView alloc] initWithFrame:CGRectMake((self.view.frame.size.width)*i, 0,self.view.frame.size.width,self.view.frame.size.height)];
            NSString *path = [NSString string];
            path = [imageBundle pathForResource:[NSString stringWithFormat:@"BBUserGuider%1d@2x",i] ofType:@"png" inDirectory:@"GuiderImage"];
            imageView.image = [UIImage imageWithContentsOfFile:path];
            [_userGuiderScrollView  addSubview:imageView];
            if (i==kBBUserGuiderImgCount-1) {
                imageView.userInteractionEnabled = YES;
                self.userGuiderImgView=imageView;
                [imageView addSubview:self.startBtn];
            }
            
            _userGuiderScrollView.contentSize=CGSizeMake((kBBUserGuiderImgCount+1)*(self.view.frame.size.width), self.view.frame.size.height);
            _userGuiderScrollView.pagingEnabled=YES;
            _userGuiderScrollView.showsVerticalScrollIndicator=NO;
            _userGuiderScrollView.showsHorizontalScrollIndicator=NO;
            _userGuiderScrollView.bounces=NO;
            _userGuiderScrollView.delegate = self;
        }
    }
    
    return _userGuiderScrollView;
}

-(UIButton *)startBtn{
    if (!_startBtn) {
        _startBtn=[[UIButton  alloc] init];
        _startBtn.backgroundColor = [UIColor clearColor];
        _startBtn.frame = self.view.frame;
        [_startBtn  addTarget:self action:@selector(startClick) forControlEvents: UIControlEventTouchUpInside];
    }
    return  _startBtn;
    
}


- (void)scrollViewDidScroll:(UIScrollView *)scrollView{
    if (scrollView.contentOffset.x > (kBBUserGuiderImgCount - 1)*(self.view.frame.size.width)) {
        [self startClick];
    }
}

//此处为 替换 rootViewController
- (void)startClick{
    [[AppDelegate shareAppDelegate] setupRootViewController:[[BBTabBarController alloc] init]];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    
}
@end

File: BBGestureBack_Full_swift/Classes/Main/BBUserGuider/BBUserGuiderController.swift
//  希望您的举手之劳，能为我点颗赞，谢谢~
//  代码地址: https://github.com/Bonway/BBGestureBack
//  BBGestureBack
//  Created by Bonway on 2016/3/17.
//  Copyright © 2016年 Bonway. All rights reserved.
//

import UIKit

class BBUserGuiderController: BBGestureBaseController {

    lazy var userGuiderScrollView: UIScrollView = {
        
        let userGuiderScrollView = UIScrollView(frame: view.bounds)
        let fileStr = Bundle.main.path(forResource: "BBUserGuider", ofType: "bundle")
        let imageBundle = Bundle(path: fileStr!)
        
        for i in 0..<kBBUserGuiderImgCount {
            let imageView: UIImageView = UIImageView.init(frame: CGRect(x: (view.frame.size.width) * CGFloat(i) ,y: 0, width: view.frame.size.width, height: view.frame.size.height))
            let path = imageBundle?.path(forResource: "BBUserGuider\(i)@2x", ofType: "png", inDirectory: "GuiderImage")
            imageView.image = UIImage.init(contentsOfFile: path!)
            userGuiderScrollView.addSubview(imageView)
            if i == kBBUserGuiderImgCount - 1 {
                imageView.isUserInteractionEnabled = true
                userGuiderImgView = imageView
                imageView.addSubview(startBtn)
            }
            
            userGuiderScrollView.contentSize = CGSize(width: CGFloat(kBBUserGuiderImgCount+1)*(view.frame.size.width), height: view.frame.size.height)
            userGuiderScrollView.isPagingEnabled = true
            userGuiderScrollView.showsVerticalScrollIndicator = false
            userGuiderScrollView.showsHorizontalScrollIndicator = false
            userGuiderScrollView.bounces = false
            userGuiderScrollView.delegate = self
        }
        return userGuiderScrollView
    }()
    
    lazy var startBtn: UIButton = {
        let startBtn = UIButton()
        startBtn.backgroundColor = UIColor.clear
        startBtn.frame = view.frame
        startBtn.addTarget(self, action: #selector(startClick), for: .touchUpInside)
        return startBtn
    }()
    var userGuiderImgView: UIImageView?
    
    
    
    override func viewDidLoad() {
        super.viewDidLoad()

        view.addSubview(userGuiderScrollView)
        
    }
    
    @objc func startClick() {
        let appDelegate = UIApplication.shared.delegate as! AppDelegate
        appDelegate.setupRootViewController(rootViewController: BBTabBarController())
    }
}

extension BBUserGuiderController : UIScrollViewDelegate {

    func scrollViewDidScroll(_ scrollView: UIScrollView) {
        if scrollView.contentOffset.x > CGFloat(kBBUserGuiderImgCount - 1)*(view.frame.size.width) {
            startClick()
        }
    }
}
